components/base/sensorcontrolled/spin.go

```diff
@@ -105,162 +52,88 @@ func (sb *sensorBase) stopSpinWithSensor(
-					sb.logger.CDebugf(ctx, "angleDeg %.2f", angleDeg)
-					sb.logger.CDebugf(ctx,
-						"currYaw %.2f, startYaw %.2f, targetYaw %.2f",
-						currYaw, startYaw, targetYaw)
-				}
+			currYaw := rdkutils.RadToDeg(orientation.EulerAngles().Yaw)
+			angErr, prevMovedAng = getAngError(currYaw, prevAngle, prevMovedAng, angleDeg)
+			prevAngle = currYaw
 
-				// poll the sensor for the current error in angle
-				// check if we've overshot our target by the errTarget value
-				// check if we've travelled at all
-				if minTravel && (atTarget || overShot) {
-					if sensorDebug {
-						sb.logger.CDebugf(ctx,
-							"stopping base with errAngle:%.2f, overshot? %t",
-							math.Abs(targetYaw-currYaw), overShot)
-					}
+			if math.Abs(angErr) < boundCheckTarget {
+				return sb.Stop(ctx, nil)
+			}
+			angVel := calcAngVel(angErr, degsPerSec)
 
-					if err := sb.Stop(ctx, nil); err != nil {
-						return
-					}
-				}
+			if err := sb.updateControlConfig(ctx, 0, angVel); err != nil {
+				return err
+			}
 
-				if time.Since(startTime) > timeOut {
-					sb.logger.CWarn(ctx, "exceeded time for Spin call, stopping base")
-					if err := sb.Stop(ctx, nil); err != nil {
-						return
-					}
-					return
+			if time.Since(startTime) > timeOut {
+				sb.logger.CWarn(ctx, "exceeded time for Spin call, stopping base")
+				if err := sb.Stop(ctx, nil); err != nil {
+					return err
 				}
+				return nil
 			}
 		}
-	}, sb.activeBackgroundWorkers.Done)
-	return nil
-}
-
-func getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errorBound float64) (atTarget, overShot, minTravel bool) {
-	atTarget = math.Abs(targetYaw-currYaw) < errorBound
-	overShot = hasOverShot(currYaw, startYaw, targetYaw, dir)
-	travelIncrement := math.Abs(angleDeg * increment)
-	// check the case where we're asking for a 360 degree turn, this results in a zero travelIncrement
-	if math.Abs(angleDeg) < 360 {
-		minTravel = math.Abs(currYaw-startYaw) > travelIncrement
-	} else {
-		minTravel = false
 	}
-	return atTarget, overShot, minTravel
 }
 
-func getCurrentYaw(ms movementsensor.MovementSensor,
-) (float64, error) {
-	ctx, done := context.WithCancel(context.Background())
-	defer done()
-	orientation, err := ms.Orientation(ctx, nil)
-	if err != nil {
-		return math.NaN(), err
+// calcAngVel computes the desired angular velocity based on how far the base is from reaching the goal.
+func calcAngVel(angErr, degsPerSec float64) float64 {
+	// have the velocity slow down when appoaching the goal. Otherwise use the desired velocity
+	angVel := angErr * degsPerSec / slowDownAng
+	if angVel > degsPerSec {
+		return degsPerSec
+	}
+	if angVel < -degsPerSec {
+		return -degsPerSec
 	}
-	// add Pi to make the computation for overshoot simpler
-	// turns imus from -180 -> 180 to a 0 -> 360 range
-	return addAnglesInDomain(rdkutils.RadToDeg(orientation.EulerAngles().Yaw), 0), nil
+	return angVel
```

Comment:
this function is what I am using to determine the velocity sent to the PID controller. The other option would be to use a trapezoidal block in the control loop and have the base configure different loops depending on what endpoint is called

Reply:
switching this to use 10% of the requested angle, with an upper bound so we do not slow down too early for long spins. Spins that are within the `boundCheckTarget`(5 degrees currently) will still end early. I should be able to reduce the `boundCheckTarget` or add a log if this case occurs if thats preferred.